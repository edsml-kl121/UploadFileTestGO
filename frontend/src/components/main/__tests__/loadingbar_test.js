import {render, screen, cleanup} from "@testing-library/react"
import Loadingbar from "../helpercomponent/loadingbar.component.js";
import renderer from 'react-test-renderer'

afterEach(() => {
  cleanup()
})


test('matches snapshot', () => {
  const tree = renderer.create(<Loadingbar props={[1,2]}/>)
  expect(tree).toMatchSnapshot();
})

test('should have expected file path', () => {
  render(<Loadingbar props ={["test.csv",2]}/>)
  const LoadingbarElement = screen.getByTestId('path-1')
  expect(LoadingbarElement).toBeInTheDocument();
  expect(LoadingbarElement).toHaveTextContent('test.csv');
})

test('should have expected progress percent', () => {
  render(<Loadingbar props ={["test.csv", 47]}/>)
  const LoadingbarElement = screen.getByTestId('progress-1')
  expect(LoadingbarElement).toBeInTheDocument();
  expect(LoadingbarElement).toHaveTextContent(47);
})
