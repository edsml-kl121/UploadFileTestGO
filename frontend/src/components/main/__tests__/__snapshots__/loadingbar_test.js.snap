// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="loadzone mb-5"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-2 col"
      >
        <img
          alt="csv"
          src="csv.png"
          style={
            Object {
              "height": "80px",
            }
          }
        />
      </div>
      <div
        className="col-lg-1"
      />
      <div
        className="col-lg-8 col"
      >
        <p
          data-testid="path-1"
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          1
          <span
            data-testid="progress-1"
            style={
              Object {
                "float": "right",
              }
            }
          >
            2
            %
          </span>
        </p>
        <div
          className="progress"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="50"
            className="progress-bar"
            role="progressbar"
            style={
              Object {
                "backgroundColor": "#6749F5",
                "width": "2%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
